tasks/lib/CodeSection.js

```diff
@@ -24,6 +24,7 @@ marked.setOptions({
     return require('highlight.js').highlightAuto(code).value;
   }
 });
+let inCodeBlock = false;
```

Comment:
don't use a static variable. Use a field in the CodeSection class instead. Initialise it in the constructor `this.inCodeBlock = false;`